test/include/data_storage.hxx

```diff
@@ -0,0 +1,99 @@
+#ifndef _STORAGE_HPP_
+#define _STORAGE_HPP_
+
+#include <cstdio>
+
+#include <RAJA/RAJA.hxx>
+
+namespace internal
+{
+
+template <typename ExecPolicy, typename T>
```

Comment:
I thought it would be convenient to have a data storage class for testing. This one is a little too specific (one or two buffers for in-place vs. out-of-place), but it can efficiently handle CPU and GPU data.